Cython/Compiler/Naming.py

```diff
@@ -183,3 +183,12 @@
 
 def py_version_hex(major, minor=0, micro=0, release_level=0, release_serial=0):
     return (major << 24) | (minor << 16) | (micro << 8) | (release_level << 4) | (release_serial)
+
+# there's a few places where it's useful to iterate over all of these
+used_macros_and_types = [
+    ('__Pyx_CyFunction_USED', [cyfunction_type_cname]),
+    ('__Pyx_FusedFunction_USED', [fusedfunction_type_cname]),
+    ('__Pyx_Generator_USED', ['__pyx_GeneratorType']),
+    ('__Pyx_IterableCoroutine_USED', ['__pyx_IterableCoroutineType']),
+    ('__Pyx_Coroutine_USED', ['__pyx_CoroutineType', '__pyx_CoroutineAwaitType'])
+]
```

Comment:
This isn't a dict. I think we can simplify this by duplicating the "USED" entries for each type cname that they guard. Why not, it's cheap. I could even imagine reversing this and putting the type cname first (and then have the option to turn it into a dict), because it seems more reasonable to look for the guard of a type than for the types that a guard protects.